test/e2e/mock-e2e.js

```diff
@@ -341,6 +356,22 @@ async function setupMocking(server, testSpecificMock) {
     });
 
   testSpecificMock(server);
+
+  // Mocks below this line can be overridden by test-specific mocks
```

Comment:
I wanted to set default mocks for these requests, but override them in the phishing tests. I found that this only worked if they were declared below this line. Presumably the first handler wins with `mockttp`.

cc @PeterYinusa , tagging you in case I'm missing something here, since you set this up.

Reply:
Thanks for tagging me here. I think I should probably add some comment blocks/documentation to provide clarity on the mocks.
Personally, I feel it needs to be cleaned up a little, as the use cases increased as we added additional test cases. Here's some general info on how it's intended to work at the moment.

1. Global mocks: Mocks in this [file](https://github.com/MetaMask/metamask-extension/blob/update-phishing-controller/test/e2e/mock-e2e.js) are supposed to be global (across all tests). These mocks are set up `before` the extension is installed.
2. Local mocks: Some mocks are specific to a single test. In this case, we use do not put them in the global mocks file, but instead place them [directly in the test](https://github.com/MetaMask/metamask-extension/blob/update-phishing-controller/test/e2e/tests/errors.spec.js#L6). Another reason we may place the mock directly in the test, is so we can do [assertions on the requests](https://github.com/MetaMask/metamask-extension/blob/update-phishing-controller/test/e2e/tests/errors.spec.js#L52).
3. Local pre-installation mocks: These are mocks that need to be in place [before the extension is installed](https://github.com/MetaMask/metamask-extension/blob/update-phishing-controller/test/e2e/tests/phishing-detection.spec.js#L102) and are specific to a single test.

Reply:
Thanks, that all makes sense.

This case still isn't clear though. Here I want to use a global mock, but override it for a specific test. Does placing it here before the invocation of the local pre-installation mocks make sense? Should we let all global mocks be overridden in this way?

Reply:
Created an issue here to tidy things up https://github.com/MetaMask/metamask-extension/issues/17843

Reply:
Actually, in reference to my comment above, we would still need a way to do assertions on a request. 
But your suggestion seems sound to me.